engine/texc/src/java/com/dynamo/bob/pipeline/TexcLibrary.java

```diff
@@ -117,4 +157,54 @@ public interface DitherType {
     public static native int TEXC_GetBufferData(Pointer buffer, Buffer outData, int maxOutDataSize);
     public static native void TEXC_DestroyBuffer(Pointer buffer);
 
+
+    private static byte[] toByteArray(BufferedImage bi, String format) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ImageIO.write(bi, format, baos);
+        byte[] bytes = baos.toByteArray();
+        return bytes;
+
+    }
+
+    private static byte[] imageToBytes(BufferedImage bi) {
+        return ((DataBufferByte) bi.getData().getDataBuffer()).getData();
+    }
+
+    private static ByteBuffer imageToByteBuffer(BufferedImage bi) {
+        return ByteBuffer.wrap(imageToBytes(bi));
+    }
+
+    private static int getPixelFormat(BufferedImage bi) {
+        switch(bi.getType())
+        {
+            case BufferedImage.TYPE_INT_RGB:        return PixelFormat.R8G8B8;
+            case BufferedImage.TYPE_INT_ARGB:       return PixelFormat.A8B8G8R8;
+            case BufferedImage.TYPE_USHORT_565_RGB: return PixelFormat.R5G6B5;
+            case BufferedImage.TYPE_BYTE_GRAY:      return PixelFormat.L8;
+        }
+        return PixelFormat.R8G8B8;
+    }
+
+    // Used for testing the importer. Usage:
+    //   ./src/com/dynamo/bob/pipeline/test_model_importer.sh <model path>
+    public static void main(String[] args) throws IOException {
+        System.setProperty("java.awt.headless", "true");
+        // TODO: Add stand alone capabilities for faster testing of apis etc
+
+        if (args.length < 1) {
+            System.out.printf("No image input specified!\n");
+            return;
+        }
+
+        File file = new File(args[0]);
+        BufferedImage image = ImageIO.read(file);
+        ByteBuffer buffer = imageToByteBuffer(image);
+        int pixelFormat = getPixelFormat(image);
+        int colorSpace = ColorSpace.SRGB;//getColorSpace(image);
+        int compressionType = CompressionType.CT_BASIS_UASTC;
```

Comment:
Maybe the configuration should be passed in as args 🤔

Reply:
Yes, it's terribly hardcoded